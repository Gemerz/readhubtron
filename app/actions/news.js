// @flow
import Moment from 'moment';
import { checkHttpStatus, parseJSON } from '../utils';

type actionType = {
  type: string,
  payload: any
};
type NewsStateType = {
  news: {
    collection: {
      length: number
    }
  }
};
export const GET_NEWS_INIT_REQUEST = 'GET_NEWS_INIT_REQUEST';
export const GET_NEWS_INIT_SUCCESS = 'GET_NEWS_INIT_SUCCESS';
export const GET_NEWS_INIT_FAILURE = 'GET_NEWS_INIT_FAILURE';

export const GET_NEWS_MORE_REQUEST = 'GET_NEWS_MORE_REQUEST';
export const GET_NEWS_MORE_SUCCESS = 'GET_NEWS_MORE_SUCCESS';
export const GET_NEWS_MORE_FAILURE = 'GET_NEWS_MORE_FAILURE';

export const SET_NEWS_LAST_CURSOR = 'SET_NEWS_LAST_CURSOR';
export const SET_NEWS_CURRENT_URL = 'SET_NEWS_CURRENT_URL';

export const CHECK_NEWS_LATEST = 'CHECK_NEWS_LATEST';


export function getNewsRequest() {
  return {
    type: GET_NEWS_INIT_REQUEST,
    payload: { loading: true }
  };
}
export function getNewsSuccess(collection: any) {
  return {
    type: GET_NEWS_INIT_SUCCESS,
    payload: { collection, loading: false }
  };
}


export function getNewsMoreRequest() {
  return {
    type: GET_NEWS_MORE_REQUEST,
    payload: { moreLoading: true }
  };
}
export function getNewsMoreSuccess(collection: any) {
  return {
    type: GET_NEWS_MORE_SUCCESS,
    payload: { collection, moreLoading: false }
  };
}
export function setNewslastCursor(lastCursor: number) {
  return {
    type: SET_NEWS_LAST_CURSOR,
    payload: { lastCursor }
  };
}
export function setNewsCurrentUrl(currentUrl: string) {
  return {
    type: SET_NEWS_CURRENT_URL,
    payload: { currentUrl }
  };
}
export function checkNewsLatest(count: number) {
  return {
    type: CHECK_NEWS_LATEST,
    payload: { count }
  };
}


export function initNews() {
  return (dispatch: (action: actionType) => void) => {
    const API = `https://api.readhub.me/news?&pageSize=10&lastCursor=${Moment().valueOf()}`;
    dispatch(getNewsRequest());
    fetch(API, {
      method: 'get',
      mode: 'cors',
      headers: {
        'Content-Length': ' application/json'
      }
    }).then(checkHttpStatus)
      .then(parseJSON)
      .then(response => {
        // dispatch(getNews(response.data));
        dispatch(getNewsSuccess(response.data));
        if (response.data[0].publishDate) {
          const lastCursor = Moment(response.data[0].publishDate).valueOf();
          dispatch(setNewslastCursor(lastCursor));
          dispatch(setNewsCurrentUrl(response.data[0].url));
        }
        return Promise.resolve(response);
      })
      .catch((e) => {
        console.log('Oops, error', e);
      });
  };
}

export function loadMoreNews() {
  return (dispatch: (action: actionType) => void, getState: () => NewsStateType) => {
    const { news } = getState();
    const cursor = news.collection[news.collection.length - 1].publishDate;
    const nextPageCursor = Moment(cursor).valueOf();
    const API = `https://api.readhub.me/news?&pageSize=10&lastCursor=${nextPageCursor}`;
    dispatch(getNewsMoreRequest());
    setTimeout(() => {
      fetch(API, {
        method: 'get',
        mode: 'cors',
        headers: {
          'Content-Length': ' application/json'
        }
      }).then(checkHttpStatus)
        .then(parseJSON)
        .then(response => {
          dispatch(getNewsMoreSuccess(response.data));
          return Promise.resolve(response);
        })
        .catch((e) => {
          console.log('Oops, error', e);
        });
    }, 500);
  };
}
export function fetchLatestCollection(lastCursor: number) {
  return (dispatch: (action: actionType) => void) => {
    const API = ` https://api.readhub.me/topic/newCount?latestCursor=${lastCursor}`;
    fetch(API, {
      method: 'get',
      mode: 'cors',
      headers: {
        'Content-Length': ' application/json'
      }
    }).then(checkHttpStatus)
      .then(parseJSON)
      .then(response => {
        dispatch(checkNewsLatest(response.count));
        return Promise.resolve(response);
      })
      .catch((e) => {
        console.log('Oops, error', e);
      });
  };
}

