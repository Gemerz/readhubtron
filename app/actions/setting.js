// @flow

export const SET_SIMPLE_LIST_MODE = 'setting/SET_SIMPLE_LIST_MODE';
export const SET_MOBILE_URL_MODE = 'setting/SET_MOBILE_URL_MODE';
export const SET_NOTIFACATION = 'setting/SET_NOTIFACATION';
export const SET_NIGNTVIEW_MODE = 'setting/SET_NIGNTVIEW_MODE';
export const SET_DISABLED_JAVASCRIPT = 'setting/SET_DISABLED_JAVASCRIPT';


export function switchSimpleMode(simpleMode: boolean) {
  return {
    type: SET_SIMPLE_LIST_MODE,
    payload: { simpleMode }
  };
}
export function switchMoblieFirstMode(moblieFirst: boolean) {
  return {
    type: SET_MOBILE_URL_MODE,
    payload: { moblieFirst }
  };
}
export function switchNotificationMode(notificationMode: boolean) {
  return {
    type: SET_NOTIFACATION,
    payload: { notificationMode }
  };
}
export function switchNightViwMode(nightViwMode: boolean) {
  return {
    type: SET_NIGNTVIEW_MODE,
    payload: { nightViwMode }
  };
}
export function switchDisabledJavascript(disabledJavascript: boolean) {
  return {
    type: SET_DISABLED_JAVASCRIPT,
    payload: { disabledJavascript }
  };
}
// export function setSimpleMode(simpleMode: boolean) {
//   return (dispatch: (action: actionType) => void, getState: () => SettingStateType) => {
//     const { setting } = getState();
//     dispatch(switchSimpleMode(simpleMode))
//   };
// }
