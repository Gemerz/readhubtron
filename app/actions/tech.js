// @flow
import Moment from 'moment';
// import type { TechStateType } from '../reducers/tech';
import { checkHttpStatus, parseJSON } from '../utils';

type actionType = {
  type: string,
  payload: any
};
type TechStateType = {
  tech: {
    collection: {
      length: number
    }
  }
};
export const GET_TECH_INIT_REQUEST = 'GET_TECH_INIT_REQUEST';
export const GET_TECH_INIT_SUCCESS = 'GET_TECH_INIT_SUCCESS';
export const GET_TECH_INIT_FAILURE = 'GET_TECH_INIT_FAILURE';

export const GET_TECH_MORE_REQUEST = 'GET_TECH_MORE_REQUEST';
export const GET_TECH_MORE_SUCCESS = 'GET_TECH_MORE_SUCCESS';
export const GET_TECH_MORE_FAILURE = 'GET_TECH_MORE_FAILURE';

export const SET_TECH_LAST_CURSOR = 'SET_TECH_LAST_CURSOR';
export const SET_TECH_CURRENT_URL = 'SET_TECH_CURRENT_URL';

export const CHECK_TECH_LATEST = 'CHECK_TECH_LATEST';


export function getTechRequest() {
  return {
    type: GET_TECH_INIT_REQUEST,
    payload: { loading: true }
  };
}
export function getTechSuccess(collection: any) {
  return {
    type: GET_TECH_INIT_SUCCESS,
    payload: { collection, loading: false }
  };
}


export function getTechMoreRequest() {
  return {
    type: GET_TECH_MORE_REQUEST,
    payload: { moreLoading: true }
  };
}
export function getTechMoreSuccess(collection: any) {
  return {
    type: GET_TECH_MORE_SUCCESS,
    payload: { collection, moreLoading: false }
  };
}
export function setTechlastCursor(lastCursor: number) {
  return {
    type: SET_TECH_LAST_CURSOR,
    payload: { lastCursor }
  };
}
export function setTechCurrentUrl(currentUrl: string) {
  return {
    type: SET_TECH_CURRENT_URL,
    payload: { currentUrl }
  };
}
export function checkTechLatest(count: number) {
  return {
    type: CHECK_TECH_LATEST,
    payload: { count }
  };
}


export function initTech() {
  return (dispatch: (action: actionType) => void, getState: () => TechStateType) => {
    const API = `https://api.readhub.me/technews?&pageSize=10&lastCursor=${Moment().valueOf()}`;
    dispatch(getTechRequest());
    fetch(API, {
      method: 'get',
      mode: 'cors',
      headers: {
        'Content-Length': ' application/json'
      }
    }).then(checkHttpStatus)
      .then(parseJSON)
      .then(response => {
        // dispatch(getTech(response.data));
        dispatch(getTechSuccess(response.data));
        if (response.data[0].publishDate) {
          const lastCursor = Moment(response.data[0].publishDate).valueOf();
          dispatch(setTechlastCursor(lastCursor));
          dispatch(setTechCurrentUrl(response.data[0].url));
        }
      });
  };
}

export function loadMoreTech() {
  return (dispatch: (action: actionType) => void, getState: () => TechStateType) => {
    const { tech } = getState();
    const cursor = tech.collection[tech.collection.length - 1].publishDate;
    const nextPageCursor = Moment(cursor).valueOf();
    const API = `https://api.readhub.me/technews?&pageSize=10&lastCursor=${nextPageCursor}`;
    dispatch(getTechMoreRequest());
    setTimeout(() => {
      fetch(API, {
        method: 'get',
        mode: 'cors',
        headers: {
          'Content-Length': ' application/json'
        }
      }).then(checkHttpStatus)
        .then(parseJSON)
        .then(response => {
          dispatch(getTechMoreSuccess(response.data));
        });
    }, 500);
  };
}
export function fetchLatestCollection(lastCursor: number) {
  return (dispatch: (action: actionType) => void, getState: () => TechStateType) => {
    const API = ` https://api.readhub.me/topic/newCount?latestCursor=${lastCursor}`;
    fetch(API, {
      method: 'get',
      mode: 'cors',
      headers: {
        'Content-Length': ' application/json'
      }
    }).then(checkHttpStatus)
      .then(parseJSON)
      .then(response => {
        dispatch(checkTechLatest(response.count));
      });
  };
}

