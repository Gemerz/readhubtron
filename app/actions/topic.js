// @flow
import { push } from 'react-router-redux';
// import type { TopicStateType } from '../reducers/topic';
import { checkHttpStatus, parseJSON } from '../utils';

type actionType = {
  type: string,
  payload: any
};
type TopicStateType = {
  topic: {
    collection: {
      length: number
    }
  }
};

type SettingStateType = {
  setting: {}
};
export const GET_TOPIC_INIT_REQUEST = 'GET_TOPIC_INIT_REQUEST';
export const GET_TOPIC_INIT_SUCCESS = 'GET_TOPIC_INIT_SUCCESS';
export const GET_TOPIC_INIT_FAILURE = 'GET_TOPIC_INIT_FAILURE';

export const GET_TOPIC_MORE_REQUEST = 'GET_TOPIC_MORE_REQUEST';
export const GET_TOPIC_MORE_SUCCESS = 'GET_TOPIC_MORE_SUCCESS';
export const GET_TOPIC_MORE_FAILURE = 'GET_TOPIC_MORE_FAILURE';

export const SET_TOPIC_LAST_CURSOR = 'SET_TOPIC_LAST_CURSOR';
export const SET_TOPIC_CURRENT_URL = 'SET_TOPIC_CURRENT_URL';
export const SET_TOPIC_TOTAL_URL = 'SET_TOPIC_TOTAL_URL';

export const CHECK_TOPIC_LATEST = 'CHECK_TOPIC_LATEST';


export function getTopicRequest() {
  return {
    type: GET_TOPIC_INIT_REQUEST,
    payload: { loading: true }
  };
}
export function getTopicSuccess(collection: any) {
  return {
    type: GET_TOPIC_INIT_SUCCESS,
    payload: { collection, loading: false }
  };
}


export function getTopicMoreRequest() {
  return {
    type: GET_TOPIC_MORE_REQUEST,
    payload: { moreLoading: true }
  };
}
export function getTopicMoreSuccess(collection: any) {
  return {
    type: GET_TOPIC_MORE_SUCCESS,
    payload: { collection, moreLoading: false }
  };
}
export function setTopiclastCursor(lastCursor: number) {
  return {
    type: SET_TOPIC_LAST_CURSOR,
    payload: { lastCursor }
  };
}
export function setTopicCurrentUrl(currentUrl: string) {
  return {
    type: SET_TOPIC_CURRENT_URL,
    payload: { currentUrl }
  };
}
export function setTopicTotalUrls(totalUrls: {}) {
  return {
    type: SET_TOPIC_TOTAL_URL,
    payload: { totalUrls }
  };
}
export function checkTopicLatest(count: number) {
  return {
    type: CHECK_TOPIC_LATEST,
    payload: { count }
  };
}


export function initTopic() {
  return (dispatch: (action: actionType) => void, getState: () => SettingStateType) => {
    const { setting } = getState();

    const API = 'https://api.readhub.me/topic?&pageSize=10';
    dispatch(getTopicRequest());
    fetch(API, {
      method: 'get',
      mode: 'cors',
      headers: {
        'Content-Length': ' application/json'
      }
    }).then(checkHttpStatus)
      .then(parseJSON)
      .then(response => {
        // dispatch(getTopic(response.data));
        dispatch(getTopicSuccess(response.data));
        if (response.data[0].order) {
          dispatch(setTopiclastCursor(response.data[0].order));
          const url = setting.moblieFirst ? response.data[0].newsArray[0].mobileUrl : response.data[0].newsArray[0].url;
          dispatch(setTopicCurrentUrl(url));
          dispatch(setTopicTotalUrls(response.data[0].newsArray));
        }
      });
  };
}

export function loadMoreTopic() {
  return (dispatch: (action: actionType) => void, getState: () => TopicStateType) => {
    const { topic } = getState();
    const nextPageCursor = topic.collection[topic.collection.length - 1].order;
    const API = `https://api.readhub.me/topic?&pageSize=10&lastCursor=${nextPageCursor}`;
    dispatch(getTopicMoreRequest());
    setTimeout(() => {
      fetch(API, {
        method: 'get',
        mode: 'cors',
        headers: {
          'Content-Length': ' application/json'
        }
      }).then(checkHttpStatus)
        .then(parseJSON)
        .then(response => {
          dispatch(getTopicMoreSuccess(response.data));
        });
    }, 500);
  };
}
export function fetchLatestCollection(lastCursor: number) {
  return (dispatch: (action: actionType) => void, getState: () => TopicStateType) => {
    const API = ` https://api.readhub.me/topic/newCount?latestCursor=${lastCursor}`;
    fetch(API, {
      method: 'get',
      mode: 'cors',
      headers: {
        'Content-Length': ' application/json'
      }
    }).then(checkHttpStatus)
      .then(parseJSON)
      .then(response => {
        dispatch(checkTopicLatest(response.count));
      });
  };
}
