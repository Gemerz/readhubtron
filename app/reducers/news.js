// @flow
import {
  GET_NEWS_INIT_REQUEST,
  GET_NEWS_INIT_SUCCESS,
  GET_NEWS_INIT_FAILURE,
  GET_NEWS_MORE_REQUEST,
  GET_NEWS_MORE_SUCCESS,
  GET_NEWS_MORE_FAILURE,
  SET_NEWS_LAST_CURSOR,
  SET_NEWS_CURRENT_URL,
  CHECK_NEWS_LATEST
} from '../actions/news';

export type NewsStateType = {
  collection: any,
  lastCursor: number,
  loading: boolean,
  currentUrl: string,
  moreLoading: boolean,
  count: number,
  category: string
};

type actionType = {
  type: string,
  payload: any
};
const initialState: NewsStateType = {
  collection: [],
  lastCursor: 0,
  loading: false,
  currentUrl: '',
  moreLoading: false,
  count: 0,
  category: 'news'
};

export default function topic(state: NewsStateType = initialState, action: actionType) {
  switch (action.type) {
    // init list
    case GET_NEWS_INIT_REQUEST:
      return {
        ...state,
        loading: action.payload.loading
      };

    case GET_NEWS_INIT_SUCCESS:
      const Collection = [];
      action.payload.collection.map((item, key) => Collection.push(item));
      return {
        ...state,
        collection: Collection,
        loading: action.payload.loading
      };

    // more list
    case GET_NEWS_MORE_REQUEST:
      return {
        ...state,
        moreLoading: action.payload.moreLoading
      };
    case GET_NEWS_MORE_SUCCESS:
      const MoreCollection = state.collection;
      action.payload.collection.map((item, key) => MoreCollection.push(item));
      return {
        ...state,
        collection: MoreCollection,
        moreLoading: action.payload.moreLoading
      };
    case SET_NEWS_LAST_CURSOR:
      return {
        ...state,
        lastCursor: action.payload.lastCursor,
      };

    case SET_NEWS_CURRENT_URL:
      return {
        ...state,
        currentUrl: action.payload.currentUrl,
      };
    // check lastest
    case CHECK_NEWS_LATEST:
      return {
        ...state,
        count: action.payload.count,
      };
    default:
      return state;
  }
}
