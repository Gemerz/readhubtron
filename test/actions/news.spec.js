import { spy } from 'sinon';
import * as actions from '../../app/actions/news';

describe('actions', () => {
  it('initNews action', () => {
    expect(actions.initNews()).toMatchSnapshot();
  });

});
