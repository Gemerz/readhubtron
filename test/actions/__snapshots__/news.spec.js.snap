// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions initNews action 1`] = `[Function]`;
